### ISSUE TITLE
Urgent subscription invoicing options not visible/dissapeared in v16+

### ISSUE THREAD

**Issue by sqvdotbe** [odoo/odoo#118858]
**Impacted versions:**
16.1+e  and higher according to current runbot versions.

**Steps to reproduce:**  
See for yourself.

**Current behavior:** 
Subscription invoicing options dissapeared after upgrading from v14 or v15 to v16, can't create any new or edit any existing subscriptions with draft invoice creation any more.
See forum: https://www.odoo.com/nl_NL/forum/help-1/odoo-v16-subcriptions-create-invoice-draft-218991

After the upgrade, any subscription with the previously active 'draft invoice' option are invoiced and sent anyway without creation of a draft invoice. This results in incorrect invoices, as the 'draft invoice' option is used to perform manual changes to invoices prior to sending them. This also results in extreme loss of time as those invoices need to be cancelled manually and checked up one by one as well as the subscriptions themselves should be pauzed, then invoices should be reissued and sent, all manually.

**Expected behavior:**
Be able to create subscriptions again with 'draft invoice' method and also the other options that were available in v14-v15.

**Support ticket number submitted via odoo.com/help (optional):**
3277339

**Comment by dalaOdoo** [comment]
Hi @sqvdotbe,
This feature has been removed in Odoo 16, subscription objects are considered to be automatic objects meaning they are not supposed to be modified at invoicing level.

Can I ask you what kind of subscriptions are you managing and why it requires manual change at invoice level ?

**WORKAROUND**
If you don't want your subscriptions to be automatically renewed, you need to pause them. 
For that you should create a specific Subscription stage (Subscriptions app > Configuration > Subscriptions stages) with category = "invoicing pause" but you will have to manually invoice them after.

**Comment by sqvdotbe** [comment]
Hi @dalaOdoo 

> This feature has been removed in Odoo 16, subscription objects are considered to be automatic objects meaning they are not supposed to be modified at invoicing level.

Yes, obviously some subscriptions of services don't require any recurring draft invoice creation, but many others do.
I can't understand that this is being removed without any announcement or warning before any upgrade.
As you can see in the forum posts, we are not alone, this makes Odoo unreliable and renders the use of Odoo unstable for use in companies, which obviously companies don't want...

The creation of draft invoices from the subscription app is frequently used by ourselves and customers, and I'm sure, many companies. For me, being able to have recurring draft invoices generated automatically at a set date was one of the major points to choose for Odoo as well as for customers. If this disappears, unfortunately we'll have to look for other alternatives to Odoo.

It's like you buy a brand new car with airbags, but after a maintenance a year later, the garage removes the airbags without prior knowledge and consent of the customer, and the customer has to find out the hard way...

> Can I ask you what kind of subscriptions are you managing and why it requires manual change at invoice level ?

At 'draft' invoice level.
In the service area we have to be able to invoice maintenance contracts which are set as a yearly or quarterly, monthly, or even  weekly subscription (for example for fixed salaries re-invoiced weekly), and at the set contract/sub renewal date, a draft invoice is created, as this draft invoice is then visible on top of the invoice list, we then know that it should be checked and requires a short manual review and edit with for example an annual price indexation or some text to be added, before the invoice is posted and sent to the customer.

Price indexation of the contract can only be done manually up to the day or month of the renewal, this is because every contract has a different price and not a fixed price, and the index values are set by the government every next month for the previous month.

So yes, next to normal automatically posted and sent invoices, we do have the need for such semi-automatic draft invoices.

> WORKAROUND
If you don't want your subscriptions to be automatically renewed, you need to pause them.
For that you should create a specific Subscription stage (Subscriptions app > Configuration > Subscriptions stages) with category = "invoicing pause" but you will have to manually invoice them after.

That's not a workaround unfortunately, as it requires manual work to check every paused subscription on a daily/weekly basis, to keep a list (with more risks of mistakes) of which contract needs to be invoiced at what time.

> There is also the possibility to create auto-post invoices. This feature will automatically post the invoice when the invoice date is reached and it will create a new draft for the next invoicing date.
I'm now aware of this possibility and tested it, but it can only be started from a new invoice, is not linked to a subscription or sales order, and limited to yearly/quarter/monthly/specific date. And after posting the invoice it blocks the editing of these settings, so when a contract without end date would end after for example letter of termination, it can't be edited and the customer will keep receiving invoices (if sent) as there is no end date set/known.

Frankly, as it was before in v14, and I'm sure others would agree, it was as good as it could be and removing it is going back to the **stone age**. I don't know who and what exactly motivated the removal of the original function, but it sure is bothering.

Since this function has been removed, no more draft invoices are created, and an accounting/invoicing responsible simply has no clue that these invoices were never made, which results in loss of revenue, invoices are not sent on time, and the customer is in fact, not covered by contract(s) any more, as he didn't get any invoices nor paid for those.

See it as having a car insurance, both parties know there is a contract, but it is never invoiced by the insurance company, and therefore you didn't pay it, resulting in 'sorry, you are not covered' or 'sorry, here are the invoices of the previous years/months', due to a software problem they weren't sent, please pay them as soon as possible. That would be a mess right?

**Comment by dalaOdoo** [comment]
@sqvdotbe thanks for your answer.
I understand your disappointment and the impact of this change on your daily work. We are already discussing internally how we could communicate better on this kind of depreciation in the future.

If I understood well, you have contracts with your clients and you invoice them periodically based on the delivered service.
- If yes, how did you manage the fact that the invoicing period mentioned on subscription invoices was the following period and not the previous one ?
- are you always creating draft invoices for all your contracts ?

**Comment by sqvdotbe** [comment]
@dalaOdoo thanks for your follow-up.

> I understand your disappointment and the impact of this change on your daily work. We are already discussing internally how we could communicate better on this kind of depreciation in the future.
I think it's necessary to communicate better on such matters, at least for Enterprise SaaS/online users as it's difficult to go (roll) back to a previous version. It was the only thing I didn't test when upgrading, and was convinced such basic functions in the Subscriptions app would remain in v16 when coming from v14.

> If I understood well, you have contracts with your clients and you invoice them periodically based on the delivered service.
Not based on a delivered product or service, the invoice is covering the period (for example: year) after the invoice date (or period mentioned in the invoice).
The services haven't been delivered yet, but are covering any needed services from the contract in the future period that is mentioned in the invoice.
See it as an omnium car insurance, every year your insurance company bills you a fee, in case your don't have any damage they don't need to do anything, in case there is, you don't have to pay anything, so'll it be, next year, same bill, but the amount is adjusted according to the price index.

Sometimes draft invoices are also needed for recurring quarterly commissions we receive from a supplier with a commission note, these commissions are often recurring with the same texts/products, but with variable amounts according to the sales of the previous quarter, which we then adjust and invoice back to the supplier(customer).

> If yes, how did you manage the fact that the invoicing period mentioned on subscription invoices was the following period and not the previous one ?
That's default behaviour in Odoo right? When you set a subscription date, the invoice will be for the period set till the next one.

> are you always creating draft invoices for all your contracts ?
No, only 2 possibilities: 

- We offer SaaS subscription products, which have a fixed price, these are invoiced and sent automatically, as no price changes are needed. IF price changes are needed, we adjust them manually on the subscription SO before the next invoice.
This is because some customers have different pricing/conditions.

- We offer service & maintenance contracts, which are invoiced automatically but as draft and adjusted before sending, because the index needs to be changed according to the variable index number which is only known up to a month before the invoice should be drafted.

**Comment by dalaOdoo** [comment]
For your information we are searching bests solutions to help customers in your case but we can not re-introduce such a feature in a stable versions.

We are currently working on:
- adding a filter in the Subscriptions module to easily identify subscriptions that needs to be invoiced: next_invoice_date <= today, so subscriptions paused that needs to be invoiced can easily be identified. 
- adding a system parameter to create all subscriptions invoices in draft (semi-automated).

Unfortunately this will not work as expected for you as you have both automated & semi-automated renewals.

Another suggestion I can do, at least to help you while we found something better is to use activity to remind that the sub must be re-evaluated based on index before it is automatically renewed.

**Comment by sqvdotbe** [comment]
@dalaOdoo 

I think the most convenient way would be to add the 'invoices in draft' parameter on a 'Recurrence period' as was the case before, but in the now so called 'Plans'. Or have the option somewhere on the sales order options

Subscriptions > Configuration > Reccurence periods

It was actually so irritating this customer created a v16 module : https://apps.odoo.com/apps/modules/16.0/beone_sub_draft_invoice/

As you can see, in my own db the recurring periods were migrated from v14 to 16 and still have the good title according to specific use (draft or automatic) and time periods. But since draft invoices option was removed, it became rather useless...

<img width="436" alt="image" src="https://user-images.githubusercontent.com/116297943/233406060-6cd4e4f6-5739-48b6-a8ad-72811d25ed6c.png">

**Comment by sqvdotbe** [comment]
Another 'thing' that would be great is an option to select add/remove (or edit the formatting) of the automatic text added behind a service description, because many already add a custom fixed date in the service description.

For example, contract/subscription date is 20/11 to 19/11 of the next year. But the invoice should be issued 01/11 of every year instead of 20/11. That means, the subscription date should be set on 01/11 and in the service description a manual date is shown.

**Comment by Yenthe666** [comment]
We also made an official support ticket for this (number 3229187) and also made other tickets about this.
In general the whole subscription flow on orders and invoices has been terrible broken in Odoo V16. Just to give some samples:
- You can no longer create partial down payment invoices when combining stockable products and subscription products
- Unable to have two subscription lines with different intervals (one monthly, one yearly)
- Documentation is completly outdated
- ...

This has been a terrible solution and has for sure not been battle tested.

**Comment by dalaOdoo** [comment]
Hi @sqvdotbe,
We re-discussed it internally and we are still convinced that re-evaluating the amount of subscriptions (or checking the quantity delivered, ... ) should be done on Subscriptions side and not on invoice side as it can impact MRR but also because it is not the job of accountants. 

We agree working with paused subscription is not good in terms of UX and so we will re-introduce a flag on a subscription plan to define if renewal must be done automatically or not, and if not, adding an alert to the sales person.

@Yenthe666,
> You can no longer create partial down payment invoices when combining stockable products and subscription products

We are working to backport the fix we made in 16.1 about this.

> Unable to have two subscription lines with different intervals (one monthly, one yearly)

It is a choice we made after having tried to manage several recurrences on the same subscription but it was too complex and very difficult to maintain. After looking at competitors, we considered it was rare and not worth to support in standard. Before it was indeed possible to create one quotation, but still it was creating different subscriptions. Now you will have to create one quotation for each subscriptions and we don't think it is a big problem.

> Documentation is completly outdated

Sorry for that, we are working on it.

**Comment by sqvdotbe** [comment]
> We re-discussed it internally and we are still convinced that re-evaluating the amount of subscriptions (or checking the quantity delivered, ... ) should be done on Subscriptions side and not on invoice side as it can impact MRR but also because it is not the job of accountants.
I understand your point of view, but we're speaking of an index of a few euros to a few hundreds of euros that need to be added on top of the previous amount already. Once that is done ON the invoice, we return to the subscription to adjust the new pricing, and do so every year. So accounting wise it doesn't really matter as on subscription level nothing is accounted for unless it has been invoiced.

FYI, even with that, the MRR shown is not calculating the set cost or purchase price of the service. It just shows a MRR based on the total price. If the subscription is yearly shown MRR is devided by 12, but the cost is not deducted.

> You can no longer create partial down payment invoices when combining stockable products and subscription products
That's a pity, I also saw this other downgrade limitation seeing the day of light.
All our quotations for a total project, include both stockable & subscription products. For some projects we propose everything in one quote.
This means that now we need to make a second separate quotation only for subscription products, resulting in more work and administration.
While before all could be confirmed and signed for by the customer at once in one quote.

**Comment by Yenthe666** [comment]
> We are working to backport the fix we made in 16.1 about this.

So in reality only odoo.com benefits :) Nobody on premise or on Odoo.sh can use this functionality. It has brought us to the point where we need to build custom apps to get this done, pissing off our customers.

> > Unable to have two subscription lines with different intervals (one monthly, one yearly)
> 
> It is a choice we made after having tried to manage several recurrences on the same subscription but it was too complex and very difficult to maintain. After looking at competitors, we considered it was rare and not worth to support in standard. Before it was indeed possible to create one quotation, but still it was creating different subscriptions. Now you will have to create one quotation for each subscriptions and we don't think it is a big problem.

"Not a big problem"... Any serious company that combines physical products with subscriptions (and surprise there's a lot) will now to make two or more quotes. You'd be serious how big that amount is.
For example: you sell charging stations combined with a monthly charging card subscription and yearly subscription for support on the station. You're already stuck and will have at least 2 quotes..

**Comment by sqvdotbe** [comment]
@Yenthe666 

> "Not a big problem"... Any serious company that combines physical products with subscriptions (and surprise there's a lot) will now to make two or more quotes. You'd be serious how big that amount is.
For example: you sell charging stations combined with a monthly charging card subscription and yearly subscription for support on the station. You're already stuck and will have at least 2 quotes..

+1

We have a customer including myself with quotation templates, in it, according to the type of sale there are optional subscription services  (yearly maintenance contract, monthly mobile Internet subscriptions or other SaaS / PaaS / IaaS subscriptions.

**Comment by dalaOdoo** [comment]
Hi @sqvdotbe, @Yenthe666 thanks for your returns this will help us improving our self for the future.

- MRR is only computed based on revenue not on cost, that how it is defined. 
- Maybe I was not clear about down payments, we did the fix in 16.1 and now we are working to put it also in 16.0
- You can still mix recurring products and non recurring products on a quotation. The recurrence is defined on the Sale Order and will apply on all recurring products in the SO but will not apply to non recurring products. But indeed as the recurrence is defined on the SO you can not mix recurring products with different recurrence periods on the same quotation. For example, you can not mix a yearly product and a monthly product on the same quotation. We know there are few businesses for which it could be useful but we decided to not cover it in standard at the moment. I created a feedback to take into account this kind of use cases for future improvements.
- Regarding the automatic generation of draft invoices, we don't plan to change the standard product back in that way as already explained. For existing customers, we are working to improve the migration process.

**Comment by Yenthe666** [comment]
> Maybe I was not clear about down payments, we did the fix in 16.1 and now we are working to put it also in 16.0

Can you link me the PR please? On my ticket Odoo gave me the finger and said I had to buy a support pack to get it custom built for our customer or wait for V17 as it would never be backported to V16. See in Odoo.com ticket number `3229187`


> We know there are few businesses for which it could be useful but we decided to not cover it in standard at the moment

Change _some_ to like 50%+ of the business world. I already have 4 projects stuck on this one.
I cannot wrap my head around how broken this now is.

**Comment by dalaOdoo** [comment]
Hello @Yenthe666, sorry for the delay but the PR was not created yet but here we are: https://github.com/odoo/enterprise/pull/40755

> Change some to like 50%+ of the business world. I already have 4 projects stuck on this one.

Can you describe me which kind of businesses exactly and why it is a blocking point ?

**Comment by codeagencybe** [comment]
> Hello @Yenthe666, sorry for the delay but the PR was not created yet but here we are: [odoo/enterprise#40755](https://github.com/odoo/enterprise/pull/40755)
> 
> > Change some to like 50%+ of the business world. I already have 4 projects stuck on this one.
> 
> Can you describe me which kind of businesses exactly and why it is a blocking point ?

Are you guys seriously really that blind or on purpose selective deaf and blind??
I really don't understand why Odoo has so much problems to understand this is the **REAL** market with real customers. 
It sounds like you hear first time about companies that sell different subscription frequencies together? I would even claim that not 50% but easy 80%+ of companies have these business cases **every single day**. 

Just like Yenthe we have also filed several complaints and tickets about this exact same problem, but everybody just gave us back the finger and leave clients with a broken Odoo on subscriptions. 
I even had customers talking about changing the entire ERP because of this problem case. 

Almost every company I know that is in the service and maintenance business have mix & match of different subscription intervals. Including my own company.

To give some real examples:
* large(s) companies that are in HVAC business which sell hardware + maintenance plans. 
Some plans are monthly, quarterly, 6-months or yearly and they are mixed together from a single quotation. 
Impossible in Odoo 16. These companies now have to create multiple quotations to get signed.
If you deal with with prepayments, it get's even more hilarious, because now the customer has to do pay multiple quotations seperately??

* hosting business, to refer also to my self. 
Domains are always sold yearly while hosting can be monthly, quarterly, up to yearly. 
Now I have to create a quotation for 1 stupid domain name seperately... 
Sometimes we sell hosting on a yearly plan with a monthly maintenance retainer. Again, impossible to combine on a single quotation. 

This situation has gone beyond amateurism from Odoo and I have no words for this to explain to clients why Odoo f... up this for no apparal reason at all. 
I totally agree that previous versions had a rough implementation with the entire subscriptions templates and often caused duplicate service products to workaround price differentation etc... 
Everybody had such big hopes that Odoo 16 finally solved that template mayhem but instead we get an even worse implementation. And all of this went silently under the radar, almost as if it was done on purpose. 

So many **breaking** changes that happen year after year. Why don't you guys don't talk first with partners about big feature changes like this? Invite partners a few times a year and talk about the upcoming roadmap so we can share experiences ahead. At least you will know the heat and problems before you actually make any big changes for the good or bad. 

Sorry for the rant, but these kind of situations really piss off clients and myself massively.

**Comment by sqvdotbe** [comment]
@codeagencybe  +1

Yeah, I can imagine the nightmare... And I'm just a starting 'learning' partner with a little voice... When I see even bigger partners have no insight or feedback channels, I remain speechless.

I have posted more tickets than I ever did for an ERP in 15 years compared to 0 tickets for our previous ERP, yes 0.

Stability, continuity, but also upgradeability of app functions to newer versions is mandatory, removing functions without replacement is a huge mistake, suddenly business processes stop working abruptly. Luckily some partners and alert direct customers are there to notice it on test databases before upgrading the production databases.

I can understand that some functions / apps need code improvement, but that shouldn’t have an extreme impact for users, if it's not ready, simply don't release it yet, or postpone the release.
On a marketing point of view, I understand the pressure to have new features released in the next version, but please, don't make a mess of it and be transparent.

Instead of migrating to Odoo I could've kept the previous ERP we were using including for our customers, and just upgrade it, it would've cost me less than posting tickets every once and a while.

'Buy a success pack' is not the answer. Stability & business continuity is.

**Comment by Yenthe666** [comment]
> 
> I have posted more tickets than I ever did for an ERP in 15 years compared to 0 tickets for our previous ERP, yes 0.
> 


You honestly don't want to know what my counter is at. I am easily over 500 tickets and gave up for the most part on reporting.
You always get back the nonsense about 'unusual busy', 'not supported', 'buy a support pack', .. 
The Odoo support has become a total nightmare. We are now at the point that we even no longer mention the free support to new customers because it is so painful that customers will come back to us saying that this is not acceptable and we as a partner have to bear it.

@codeagencybe I could not agree with you more. Odoo its professionalism seems to go down year by year and has hit absolute lows. Odoo as a software is marvelous because of the flexibility and what it can bring.. but the company Odoo SA? Total nightmare to work with. Even if there is a typo in a customer it's name and I email my success manager / AM they tell me that I have to make a support ticket. Like, wtf?

@fpodoo I think it is time you start listening to your partners. I see more and more of them getting frustrated every single day.

**Comment by collaborativeandreas** [comment]
don't know if someone from odoo is still inerrested in this topic... but i can fully confirm all customer / partner opinions in this threat...  odoo messed up the subscription module by removing the draft invoice option (or wants to force all customers to custom apps and more expensive subscriptions) 

And removing such an important feature without a upfront warning or something else is a nightmare and increased our invoicing efforts massively! (we are selling subscriptions, hardware and maybe addidtional time on a single Monthly invoice) 

and as @Yenthe666 stated: odoo support became a nightmare... long resonse times, no clear answers, no support for partners, and always having to explain to the customers why something isn't working as expected  - and especially in this topic no clear statement on how to solve it... waitung for V17 (and hoping for a fix in V17) isn't a real option

**Comment by Yenthe666** [comment]
> The only thing that happened is that it got closed silently without updated feedback.

Lately that seems to be the case on loads of subjects sadly. Dissapointing evolution and also why I decided to no longer contribute to the Odoo core.

**Comment by collaborativeandreas** [comment]
Glad to have this confirmed... but there will be no solution for Odoo 16?
Don't know if I ever want to upgrade odoo again ... or how i should explain it to our customers

**Comment by dalaOdoo** [comment]
@collaborativeandreas there are solutions depending on the customer use case but indeed it will imply changes in the way the customer is working. 

If you are looking for a solution with Odoo 16 working exactly as Odoo 15 and no impact on the customer then the only solution is customization.

**Comment by dalaOdoo** [comment]
Believe it or not, I have nothing to sell...

**Comment by dalaOdoo** [comment]
Sorry if my messages were unclear but I was not inviting you to discuss about customizations. 

I wanted to say that we do believe there are standard solutions for your customers in 16 but of course, as Odoo changed a lot between 15 and 16, it will require to change the way of working.
As I don't know your customers as much as you do, I wanted to invite you to discuss their use cases and if indeed, it is not possible to find solutions for your customers with Odoo 16 then we will be able to point out what is missing in order to evaluate if it can be done for Odoo 17 or not.

If you are interested in such a talk then don't hesitate to contact me by email.

**Comment by codeagencybe** [comment]
> Sorry if my messages were unclear but I was not inviting you to discuss about customizations.
> 
> I wanted to say that we do believe there are standard solutions for your customers in 16 but of course, as Odoo changed a lot between 15 and 16, it will require to change the way of working. As I don't know your customers as much as you do, I wanted to invite you to discuss their use cases and if indeed, it is not possible to find solutions for your customers with Odoo 16 then we will be able to point out what is missing in order to evaluate if it can be done for Odoo 17 or not.
> 
> If you are interested in such a talk then don't hesitate to contact me by email.

I have sent you an email

**Comment by isofterp** [comment]
We're in pretty much the same boat as many others. We cannot consider V16 as a reliable platform and skipping it all together. I am now stuck on v15 with subscriptions as it works for us. V16 Subscriptions is a watered-down, thoughtless piece of crap and thats putting it mildly. It seems with every new version, someone comes up with a stupid idea to remove functionality that has a better success rate of being adopted and usuable. Why Odoo SA don't invite their partners and community before making these decisions is beyond me, but leave us to sell the product.

**Comment by codeagencybe** [comment]
> We're in pretty much the same boat as many others. We cannot consider V16 as a reliable platform and skipping it all together. I am now stuck on v15 with subscriptions as it works for us. V16 Subscriptions is a watered-down, thoughtless piece of crap and thats putting it mildly. It seems with every new version, someone comes up with a stupid idea to remove functionality that has a better success rate of being adopted and usuable. Why Odoo SA don't invite their partners and community before making these decisions is beyond me, but leave us to sell the product.   


Simple, they just don't care. 
6 months after so many complaints about the changes in v16 there is nothing done to address this issue in v17. 
So in a few weeks they just shove the same crap again and ignore all clients who are in shit. 

Keep in mind that all clients that are on odoo.sh will have no option to avoid upgrade. 
Starting v18, all those clients on odoo.sh will auto upgrade to v16 minimum. 
Chaos countdown can start...

**Comment by collaborativeandreas** [comment]
btw... are there any known rumors about the subscription module in V17? 
@codeagencybe  did you ever received an answer on your mail to @dalaOdoo  in June? Just interested because I never received an answer on my mail and we're still working with "paused" subscriptions :-(

**Comment by dalaOdoo** [comment]
@collaborativeandreas when did you contact me ? I cant' find your email

**Comment by gabriumaa** [comment]
Today I discovered this Odoo error.
I already have two angry clients.

For the moment, I will take @sqvdotbe's solution with its module.

**Comment by ortnerthomas** [comment]
We have the same problem. We have Sales Orders with stock items and subscriptions items with different recurrence periods almost daily. And i really can not send one customer 3 sales orders because Odoo is not able to fix this problem. I also need the function back that invoices are not send automatic but in drafts, since i also have to check it and add some text before sending the invoice. 

I would also like the additional feature for different invoice date then subscription dates. Because we also send the invoice a month before the actuall subscription expiration date for some of our customers.

So is there any news on this post?

**Comment by sqvdotbe** [comment]
@dalaOdoo any update on this? This was actually really the only feature which made me decide to move to Odoo from another ERP. It made me win 80% of my time by removing manual work, now I need to do this manual work again for already more than 8 months.
I would like to get this time back so I can spend it with my family or other more important things instead of spending it in front of my screen…

@fpodoo As you said in the Odoo Experience 2023 'Your Questions, Our Answers: Ask Me Anything' maybe decision making should also be reviewed before employees make unilateral decision to remove highly used top features like this.

**Comment by codeagencybe** [comment]
> @codeagencybe
> 
> > Also didn't change or improve in v17, they just kept the same borderline horrible experience.
> 
> Are you serious? So we still have no solution to this sh*tshow?!

Nope, it got worse from my POV. 
v17 now supports recurring deliveries for stockable products. That's a nice feature, but the user flow and UX is horrible. 
If you create a quotation with goods or services you sent once and with a recurring plan, then *everything* becomes a subscription. 
You can't say this is one time and the rest is recurring/subscription. It's all or nothing. 
So again, you have to create seperate SO to make sure they don't get mixed up. 

So the problem from v16 only now got worse because stockable products are now also involved in the mess.

**Comment by dalaOdoo** [comment]
We decided to not support anymore since Odoo 16:
- The automatic generation of draft invoices and we do not plan to change that. If a user wants to work in a manual flow to check/modify/send invoices, he has to put subscriptions in pause (or stop the scheduled actions). Odoo will no more take care of these subscriptions automatically. For the user, creating all the draft invoices takes a few number of clicks in comparison to all the clicks to check, modify and send invoices manually. The user has to filter subscriptions 'to invoice', then select all and click on 'create invoice'.
- The multi recurrence on the same quotation to create multiple subscriptions: we tried in a first place and it was a nightmare technically and functionally for users. Most subscriptions management softwares do no support it either (chargebee, recurly, billsby, ... ). We do not plan to support within the subscriptions app, the re-invoicing of external "subscriptions" (server hosting, software licences, ...). Usually in these kind of use cases, the subscriptions are started later (when the project is done, when the licences has been created, ...). 
 
@codeagencybe regarding subscription of storable products: only recurring products will be invoiced and delivered recurrently. If you have 'simple' products in your SO, it will be invoiced and delivered only once.

CC @ortnerthomas @Yenthe666 @sqvdotbe

**Comment by Yenthe666** [comment]
> Most subscriptions management softwares do no support it either (chargebee, recurly, billsby, ... ).

Wasn't the goal of Odoo to not be like others and do better where others stop? 
Some of the choices made here are beyond stupid. Heck, some features are even nerfed/killed compared to version 15.
Sorry, it simply baffles me what a short sighted implementation and choices where made here.

**Comment by codeagencybe** [comment]
> * The multi recurrence on the same quotation to create multiple subscriptions: we tried in a first place and it was a nightmare technically and functionally for users. Most subscriptions management softwares do no support it either (chargebee, recurly, billsby, ... ). We do not plan to support within the subscriptions app, the re-invoicing of external "subscriptions" (server hosting, software licences, ...). Usually in these kind of use cases, the subscriptions are started later (when the project is done, when the licences has been created, ...).
> 

But it never was a problem in older versions of Odoo? There was nothing wrong with this feature or functionality. 
And besides, there are many other subscription platforms that do it just fine. 
You are just making up excuses for the poor underdeliverable from Odoo. 
All the signs are so clear this was a rushed unfinished version that a lot of people neglected the impact on real companies their business. 

At this moment, I have 7 out of 10 customers that are seriously considering to leave Odoo because from v16 it completely destroyed their entire business model they had working just fine for the 5+ years. 
And the problem is not so much the fact it changed, but the fact that Odoo SA makes these drastic breaking changes year over year without even consulting it's real user base to understand what conflicts it causes for companies. 
There are dozens of other possibilties that could have been done to avoid this situation:
* only fix the older subscription template part without breaking the rest
* at least first checked with real customers what kind of impact it has (99,99% of companies in any type of service business now has a problem!!)
* you could have implemented options where companies decide if they want the new behaviour or not.

**Comment by dalaOdoo** [comment]
@codeagencybe

> But it never was a problem in older versions of Odoo?

In the new approach of working with the SO becoming the subscription, it was too complex making it works in all cases and keeping it simple at the same time. Most of the time if you are providing a subscription service of your own (netflix, spotify, food delivery, ...) you don't mix-up periodicities. If you do, it's probably subscriptions that will not start right away when the customer accepts the offer because it will be linked to a service(s) you need to provide first (IT projects, providing software licences, installation of a machine, ... ). 

> Some products can be sold both one time and recurring. [...] both scenario's are frequent and common, but since Odoo 16 it's impossible.

I don't know how your were managing that with only one product before Odoo 16 as you needed one product per invoicing period. We consider that a subscription and a one time product are not the same in your catalog but We agree it can be the same physical product in your stock and for that you can use a "kit" if you need.

**Comment by kbb01** [comment]
Hi

The question is does odoo even recognise the pain this has caused and are
they addressing the issue, from what I see, they aren't :(

**Comment by codeagencybe** [comment]
> @codeagencybe
> 
> > But it never was a problem in older versions of Odoo?
> 
> In the new approach of working with the SO becoming the subscription, it was too complex making it works in all cases and keeping it simple at the same time. Most of the time if you are providing a subscription service of your own (netflix, spotify, food delivery, ...) you don't mix-up periodicities. If you do, it's probably subscriptions that will not start right away when the customer accepts the offer because it will be linked to a service(s) you need to provide first (IT projects, providing software licences, installation of a machine, ... ).
> 
> > Some products can be sold both one time and recurring. [...] both scenario's are frequent and common, but since Odoo 16 it's impossible.
> 
> I don't know how your were managing that with only one product before Odoo 16 as you needed one product per invoicing period. We consider that a subscription and a one time product are not the same in your catalog but We agree it can be the same physical product in your stock and for that you can use a "kit" if you need.

No you are just making wrong assumptions. 
You didn't consult partners, you didn't consults your existing customer base, you just made the changes and enforce everybody this bad implementation. period.

The only thing I hear is exactlyu what @kbb01 also says: excuses, excuses, excuses, ...
Odoo seems uncapable of accepting and telling honestly they screwed up a major feature that existed in so many Odoo versions. 
Is it complex? Yes, sure but so is accounting and inventory and .... 
Subscriptions was not "optimal" in v15 and below, but what you guys did in v16 and onwards just made it worse. 

I really for the sake of life don't understand why you guys keep staying blind for this pain problem for so many clients. All we keep hearing is excuses and BS instead of apologizes and more important to fix it asap because it's not impossible. 

If people like @sqvdotbe and me are capable to to build fixes around the mess your team build, I don't understand why Odoo itself can not.

**Comment by michaelhofer-slg** [comment]
After following this thread and migrating to Odoo 17, we are at a point where we have lots of confirmed susbscriptions starting on January 1st next year. We do **NOT** want Odoo to send automatically invoices to all the customers, we want to review the invoice first and then send it manually.

To avoid a nightmare on New Years Day, is it enough to disable the automated action as @dalaOdoo mentioned?

> The automatic generation of draft invoices and we do not plan to change that. If a user wants to work in a manual flow to check/modify/send invoices, he has to put subscriptions in pause (**or stop the scheduled actions**)

I assume it's the scheduled action called `Sale Subscription: generate recurring invoices and payments` which invokes `model._cron_recurring_create_invoice()` on the sale order model?

Has this any other side effects or will this give us the option to create the invoices manually?

**Comment by michaelhofer-slg** [comment]
> > @codeagencybe regarding subscription of storable products: only recurring products will be invoiced and delivered recurrently. If you have 'simple' products in your SO, it will be invoiced and delivered only once.
> 
> Some products can be sold both one time and recurring. If I run a coffee shop, I maybe have a product that is available as a one-time purchase like any normal eshop, but optionally customers may want to have it as a weekly subscription. So in this case, it's the one and same product that is used for both. If I have 2 products like that can be used onetime and subscription, and I use 1 of them with intention as one time and the other one as recurring, then both become recurring.
> 
> If I sell a service, eg maintenance in hours, I can sell it onetime a number of hours but I can also sell it as a subscription for X hours weekly/monthly. Maybe I'm creating a quotation where I need some maintenance that is required one time only but another one that recurs weekly.
> 
> both scenario's are frequent and common, but since Odoo 16 it's impossible. I'm either now forced to duplicate my products/services to have explicit onetime and recurring ones (so now we are back in the subscription templates anyway!! -> so you didn't improve anything in the end!!
> 
> Or I'm forced to sent multiple quotations, multiple signs, multiple payments by clients, multiple administration to follow up, everything became less efficient and obnoxious.
> 
> Either way, Odoo has completely killed all the efficiency for companies that deal with above cases, and this is a huge part of industry.

+1

We have the situation that we sell licenses with a fixed price as one-off purchase, but in order to get updates after the first year, a yearly subscription of 15 / 20 % is required. This is currently impossible to manage without having two separate products (one with recurring prices, one without) - so we need two different products for the same item. We were told once that this will "greatly improve" in Odoo 17, but unfortunately, there was no improvement at all here.

**Comment by codeagencybe** [comment]
> After following this thread and migrating to Odoo 17, we are at a point where we have lots of confirmed susbscriptions starting on January 1st next year. We do **NOT** want Odoo to send automatically invoices to all the customers, we want to review the invoice first and then send it manually.
> 
> To avoid a nightmare on New Years Day, is it enough to disable the automated action as @dalaOdoo mentioned?
> 
> > The automatic generation of draft invoices and we do not plan to change that. If a user wants to work in a manual flow to check/modify/send invoices, he has to put subscriptions in pause (**or stop the scheduled actions**)
> 
> I assume it's the scheduled action called `Sale Subscription: generate recurring invoices and payments` which invokes `model._cron_recurring_create_invoice()` on the sale order model?
> 
> Has this any other side effects? I assume the subscription will remain active, but just not invoiced automatically?

That should do the trick yes. It will just block the invoices going out without affecting the subscription period. 
At least, that's how it's done in v15 and below also. 
In v17 there is also a new option to pause subscriptions but I don't know if that affects the subscription period. 
If it doesn't, you could also bulk pause, make the changes, resume and then the invoices will shoot again after resuming

Or you can check the module from @sqvdotbe mentioned above which will give you a new option to generate draft invoices. I don't know if it's v17 yet, but Ken can confirm this for you.

**Comment by codeagencybe** [comment]
> > > @codeagencybe regarding subscription of storable products: only recurring products will be invoiced and delivered recurrently. If you have 'simple' products in your SO, it will be invoiced and delivered only once.
> > 
> > 
> > Some products can be sold both one time and recurring. If I run a coffee shop, I maybe have a product that is available as a one-time purchase like any normal eshop, but optionally customers may want to have it as a weekly subscription. So in this case, it's the one and same product that is used for both. If I have 2 products like that can be used onetime and subscription, and I use 1 of them with intention as one time and the other one as recurring, then both become recurring.
> > If I sell a service, eg maintenance in hours, I can sell it onetime a number of hours but I can also sell it as a subscription for X hours weekly/monthly. Maybe I'm creating a quotation where I need some maintenance that is required one time only but another one that recurs weekly.
> > both scenario's are frequent and common, but since Odoo 16 it's impossible. I'm either now forced to duplicate my products/services to have explicit onetime and recurring ones (so now we are back in the subscription templates anyway!! -> so you didn't improve anything in the end!!
> > Or I'm forced to sent multiple quotations, multiple signs, multiple payments by clients, multiple administration to follow up, everything became less efficient and obnoxious.
> > Either way, Odoo has completely killed all the efficiency for companies that deal with above cases, and this is a huge part of industry.
> 
> +1
> 
> We have the situation that we sell licenses with a fixed price as one-off purchase, but in order to get updates after the first year, a yearly subscription of 15 / 20 % is required. This is currently impossible to manage without having two separate products (one with recurring prices, one without) - so we need two different products for the same item. We were told once that this will "greatly improve" in Odoo 17, but unfortunately, there was no improvement at all here.

Everybody is affected one way or another but in a bad way. 
Since v16 it's the worst possible implementation they could have made. Any company that rely heavy on subscriptions, should not upgrade past v15 or they will regret it so hard. 

I'm sticking with v15 on purpose just for this mess until I have my working custom module ready to workaround their mistakes. 

And still it remains silence in all possible ways from Odoo. They just refuse to acknowledge they made a mistake and don't want to rectify anything in any possible way. 
And that's the biggest problem I have with this.

**Comment by sarunas-devoro** [comment]
Absolute disgrace, whole subscription module is no longer usable, we always used to generate invoices for our monthly services and the add all the add hoc charges that we gathered through the month, also we charge for the previous month at the end of te month now if we leave them to be send automatically they even add text that it is for up coming month which makes it a false information so we do all the work manually... either this needs to be changes or we and some of our customers will have to ditch the module or look for alternatives.

**Comment by moylop260** [comment]
FYI a case of use broken

https://github.com/odoo/odoo/issues/156843

**Comment by gulosito** [comment]
Absolutely disappointed Odoo that you make us work the way you want, especially after being able to do it like that in v14... and suddenly because some Belgian thinks of changing the way we all work... outrageous!!!!! !!

**Comment by droidosergioh** [comment]
Is there any module or something that really solves this problem? It is just as simple as adding the old "Generate Draft Invoice" and "Manually" that was already working 2 years ago in v15

**Comment by codeagencybe** [comment]
> Is there any module or something that really solves this problem? It is just as simple as adding the old "Generate Draft Invoice" and "Manually" that was already working 2 years ago in v15

There is a module from @sqvdotbe here: https://apps.odoo.com/apps/modules/16.0/beone_sub_draft_invoice/

**Comment by Yenthe666** [comment]
> > Is there any module or something that really solves this problem? It is just as simple as adding the old "Generate Draft Invoice" and "Manually" that was already working 2 years ago in v15
> 
> There is a module from @sqvdotbe here: https://apps.odoo.com/apps/modules/16.0/beone_sub_draft_invoice/

The fact that it is already bought 86 times while it is barely findable and has like zero documentation/sexy sales page tells you enough 😓

**Comment by sqvdotbe** [comment]
Hey guys, stop mentioning me, this is not my module at all! I can't even program except write the PHP echo line 🤣 

Btw. Another one of those big screw ups by Odoo is the POS loyalty program. They managed to break it rendering it unusable in v17.2. Slowing down the entire app, almost rendering it unusable too. Imagine POS employees in a shop with a slow POS application (computer is totally fine and fast!).

Previous Monday eveningI had to do an upgrade of an Odoo Online SaaS v17.1 POS customer to v17.2 to solve another issue with POS reports which broke and produced an error message. After testing the upgraded db this issue was solved and POS seemed to work fine when testing it. I then proceeded to backup the v17.1 production db and upgrade it to v17.2, then went to sleep.

Well, by the morning the customer had to close his shop for 3 or more hours (9 to 12ish) while Odoo support was trying to find the issue because the POS sessions couldn't be started due to the loyalty program failure. I had to disable the loyalty program and now 6000+ customers don't receive their points any more resulting in them to have to print the tickets (double sometimes) to later add the points manually to every customer loyalty card.

I'm really losing it @fpodoo , stop the upgrades and upgrades every 2-3 months and please concentrate on QA, long term support, stability + great documentation. Don't save money on QA, it's money well spent. My wife is QA automation engineer in a huge software dev company which works with multi-nationals, and this is a necessary process within every software company, it seems there isn't any QA at all at Odoo.
If this continues, the rush to upgrade will be the biggest problem of Odoo stalling instead of growing.
And this is just the tip of the iceberg as a small-scale learning partner with only 5 customers. I can't imagine how bigger partners have to handle such mess with big customers.

**Comment by Yenthe666** [comment]
@sqvdotbe a quick tip. Never go with minor releases like 17.1, 17.2 and never go with Odoo.com.
It brings you at the forefront of being beta-tested and you get constant updates/upgrades. At least stick to major version (V16, V17), it will bring you more peace and less hassle. Go for major versions with Odoo.sh or own hosting, so you have much more flexibility and control.

**Comment by michaelhofer-slg** [comment]
> @sqvdotbe a quick tip. Never go with minor releases like 17.1, 17.2 and never go with Odoo.com. It brings you at the forefront of being beta-tested and you get constant updates/upgrades. At least stick to major version (V16, V17), it will bring you more peace and less hassle. Go for major versions with Odoo.sh or own hosting, so you have much more flexibility and control.

I totally agree - this gives much more control and stability. We are very happy with this solution. We usually wait 4-6 months after a new major version and upgrade after that.

Nonetheless, a solution to the issue above would be much appreciated!
